app/controllers/app.go

```diff
@@ -0,0 +1,142 @@
+	package controllers
+
+import (
+	"github.com/revel/revel"
+	"fmt"
+	_ "github.com/go-sql-driver/mysql"
+	"golang.org/x/crypto/bcrypt"
+	"github.com/gocql/gocql"
+	"io/ioutil"
+	"encoding/json"
+)
+type Responseobject struct {
+	In bool
+	Error string
+	Userdata UserContancts
+	}
+type App struct {
+	*revel.Controller
+	db *gocql.Session
+
+}
+
+type Userlogins struct{
+	username string
+	password string
+}
+func (c App) Index() revel.Result {
+
+
+	return c.Redirect("/app/index.html")
+}
+
+
+func (userinfo * Userlogins) CheckUsernameExists( db *gocql.Session) error{
+	var databasePassword string
+
+	err := db.Query("SELECT password FROM user_logins WHERE username=?", userinfo.username).Scan( &databasePassword)
+	fmt.Println(err)
+	return err
+}
+
+func (userinfo * Userlogins) QueryUser(db *gocql.Session) (string,error){
+	var databasePassword string
+
+	err := db.Query("SELECT password FROM user_logins WHERE username=?", userinfo.username).Scan( &databasePassword)
+	return databasePassword,err
+}
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+func (userinfo * Userlogins) InsertUser( hashedPassword []byte ,db *gocql.Session) error{
+	err :=db.Query("INSERT INTO user_logins(username, password) VALUES(?, ?)",userinfo.username, hashedPassword).Exec()
+	return err
+
+}
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+func (c App) Signin() revel.Result{
+	c.Params.Get("user")
+		userlogins := map[string]string{}
+		content, _ := ioutil.ReadAll(c.Request.Body)
```

Comment:
remember to discuss this with mostafa cherri as i think this is the only way to recieive JSON objects, we can use binding of revel only  when it's sending Content-Type of data as Form-data not JSON objects.